src/database/DataTypes/StocktakeBatch.js

```diff
@@ -211,7 +211,25 @@ export class StocktakeBatch extends Realm.Object {
    */
   set countedTotalQuantity(quantity) {
     // Handle packsize of 0.
-    this.countedNumberOfPacks = this.packSize ? quantity / this.packSize : 0;
+    const countedNumberOfPacks = this.packSize ? quantity / this.packSize : 0;
+
+    this.countedNumberOfPacks = countedNumberOfPacks;
+    this.doses = countedNumberOfPacks * this.dosesPerVial;
+  }
```

Comment:
when changing the counted number of packs, just adjust the number of doses issued to what it 'should' be.. i.e. if the counted number of vials is 10, and each vial has 2 doses, set it to 20. Basically makes the user enterign details have an easier time. They can adjust it more seperately